src/beast/beast/nudb/store.h

```diff
@@ -195,6 +162,13 @@ class basic_store
     std::thread thread_;
     std::condition_variable_any cond_;
 
+    // These allow insert to block, preventing the pool
+    // from exceeding a limit. Currently the limit is
+    // baked in, and can only be reached during sustained
+    // insertions, such as while importing.
+    std::size_t commit_limit_ = 1UL * 1024 * 1024 * 1024;
```

Comment:
If `commit_limit_` is truly baked in you could declare it as `static const` and get a performance advantage.  Your choice.